src/coreclr/build-runtime.cmd

```diff
@@ -392,9 +392,19 @@ if %__BuildNative% EQU 1 (
         set __ExtraCmakeArgs="-DCMAKE_BUILD_TYPE=!__BuildType!"
     )
 
-    set __ExtraCmakeArgs=!__ExtraCmakeArgs! "-DCLR_CMAKE_TARGET_ARCH=%__TargetArch%" "-DCLR_CMAKE_TARGET_OS=%__TargetOS%" "-DCLI_CMAKE_FALLBACK_OS=%__HostFallbackOS%" "-DCLR_CMAKE_PGO_INSTRUMENT=%__PgoInstrument%" "-DCLR_CMAKE_OPTDATA_PATH=%__PgoOptDataPath%" "-DCLR_CMAKE_PGO_OPTIMIZE=%__PgoOptimize%" %__CMakeArgs%
-    echo Calling "%__RepoRootDir%\eng\native\gen-buildsys.cmd" "%__ProjectDir%" "%__IntermediatesDir%" %__VSVersion% %__HostArch% %__TargetOS% !__ExtraCmakeArgs!
-    call "%__RepoRootDir%\eng\native\gen-buildsys.cmd" "%__ProjectDir%" "%__IntermediatesDir%" %__VSVersion% %__HostArch% %__TargetOS% !__ExtraCmakeArgs!
+    set __ExtraCmakeArgs=!__ExtraCmakeArgs! "-DCLR_CMAKE_TARGET_ARCH=%__TargetArch%" "-DCLR_CMAKE_TARGET_OS=%__TargetOS%"
+    set __ExtraCmakeArgs=!__ExtraCmakeArgs! "-DCLI_CMAKE_FALLBACK_OS=%__HostFallbackOS%" "-DCLR_CMAKE_PGO_INSTRUMENT=%__PgoInstrument%" "-DCLR_CMAKE_OPTDATA_PATH=%__PgoOptDataPath%" "-DCLR_CMAKE_PGO_OPTIMIZE=%__PgoOptimize%"
+
+    if /i "%__TargetOS%" == "android" (
+        if not "%__HostOS%" == "" (
+            set "__TargetOS=!__HostOS!"
```

Comment:
If command line says `-os android -hostos windows`, this line is going to override the requested configuration to `-os windows -hostos windows`. Why would we want to do that? If the caller of the script asks for android targeting binary, we should build an android targeting binary and not ignore what the command line asked for.

Reply:
Yeah, I think that was another part of the workaround. I'll remove it and see.

Reply:
> https://github.com/dotnet/runtime/blob/457e3eb93e35f60f9b0515ee6a4b75e76a64ce8c/eng/native/build-commons.sh#L521-L523

This condition makes sense. It sets HostOS to TargetOS when the HostOS was not specified (ie it sets the default). Can we do it the same way here? Set the default if the value is not specified instead of overriding what was specified on the command line.